src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -117,7 +117,7 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync(string userFlow)
             var accounts = await GetAccountsInternalAsync(ApiIds.GetAccountsByUserFlow).ConfigureAwait(false);
 
             return accounts.Where(acc =>
-                acc.HomeAccountId.ObjectId.Split('.')[0].EndsWith(
+                acc.HomeAccountId.ObjectId.EndsWith(
```

Comment:
I don't see the reason for adding this split here as the ObjectId is obtained after splitting the homeAccountId with the `.`